ooaklee/ghatd
# [GHATD-X] Implement CLI command - new

### Context:

**Related Link(s)**: `N/A`


### What has been done:

Updates include:

- Initial steps added to `new` command
- Updates to error key, Detail config, and `go.mod` tags
- "New" command logic for restructuring and merging details into a new app
- Refactor toolbox can search/replace strings, *DetailEmbeds removed
- Output added to the available `new` flag, "web" suffix added to embeddedContentFilePathPrefix
- Fetch new app's core from GitHub; clean-up triggered on failure
- Template and ADR added for converting the project to Detail and modularized set-up
- Access manager, API token manager, audit logger, auth, emailer, ephemeral data storage, middleware, OAuth manager, user manager, and user management service added to help users get started
- Repository methods added for DB writing services; common headers, static User Roles, and toolbox helper functions added
- Access manager handler now extendable; method added for cleaning all tokens except current; auth service updated
- Paginator shapes pagination response; title case added to toolbox; helper functions added to repository package to calculate pagination skip and limit
  - A method was added to get total users based on parameters.

#### Things to do later
- Update the `new` command to support private repositories 

### How to test:

- Clone repo locally
- check out this branch `ghatd-x-implement-cli-new-command`
- Run the following command:
```sh
go run cli/cli.go new -n "awesome-service" -w "github.com/ooaklee/ghatd-detail-web-demo-landing-dash-and-more,github.com/ooaklee/ghatd-detail-api-demo-endpoints"
```
- Go to the directory referenced by the cli output message saying "path to new service..."
- Once inside, run `go mod tidy`
- Verify that the new service starts by running `go run main.go start-server`
